File: Backend/mapping/src/main/java/com/mhp/coding/challenges/mapping/models/db/blocks/TextBlock.java
package com.mhp.coding.challenges.mapping.models.db.blocks;

public class TextBlock extends ArticleBlock {

    private String text;

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
    
    @Override
    public com.mhp.coding.challenges.mapping.models.dto.blocks.TextBlock map() {
        
        com.mhp.coding.challenges.mapping.models.dto.blocks.TextBlock blockDto
          = new com.mhp.coding.challenges.mapping.models.dto.blocks.TextBlock();

        blockDto.setSortIndex(this.getSortIndex());
        blockDto.setText(this.getText());
        
        return blockDto;
    }
}

File: Backend/mapping-kotlin/src/main/kotlin/com/mhp/coding/challenges/mapping/models/db/blocks/TextBlock.kt
package com.mhp.coding.challenges.mapping.models.db.blocks

class TextBlock(
    var text: String,
    override var sortIndex: Int = 0,
) : ArticleBlock(sortIndex) {

    override fun map():
        com.mhp.coding.challenges.mapping.models.dto.blocks.TextBlock {
    
        val textBlockDto =
            com.mhp.coding.challenges.mapping.models.dto.blocks.TextBlock(
        
            text = this.text,
            sortIndex = this.sortIndex
        )    
        
        return textBlockDto
    }
}
